libsplinter/src/circuit/template/rules/set_metadata.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::super::yaml_parser::v1;
+use super::Value;
+
+pub(super) struct SetMetadata {
+    metadata: Metadata,
+}
+
+impl From<v1::SetMetadata> for SetMetadata {
+    fn from(set_metadata: v1::SetMetadata) -> Self {
+        SetMetadata {
+            metadata: Metadata::from(set_metadata.metadata().clone()),
+        }
+    }
+}
+
+pub(super) enum Metadata {
+    #[allow(dead_code)]
+    // This enum is never directly build only converted from the Yaml version,
+    // #[allow(dead_code)] avoids lint warnings about dead code.
```

Comment:
If you switch the `From<v1::Metadata>` implementation below to create the variant using `Metadata::Json` instead of `Self::Json`, this lint warning should go away (weird, I know!).